### PR TITLE
Addressing RFC8776-bis WG LC comments

### LINKED ISSUE #281: Clarify why path-computation-error-reason has not been defined in an IANA-maintained model
> Offload path-computation-error-reason to an IANA-maintained model. The current design is suboptimal. Please refer to the 84047bis on these matters.

### LINKED ISSUE #284: Other formats for the MPLS-TE path bandwidth
There is this text in section 3.3:

```
   NOTE: Other formats for the MPLS-TE path bandwidth are defined in
   [I-D.ietf-teas-yang-te-mpls] and they could be added in a
   future update of this document.
```

### LINKED ISSUE #285: Bandwidth profile groupings

### LINKED ISSUE #287: Check whether informative references should be normative
> I think that many informative references should move to be normative. For example, RFC6370 should be normative. Please double check.
>

### LINKED ISSUE #288: Simplify unions with overlapping types

### LINKED ISSUE #290: Some Med's Minor WG LC comments
> > # Minor comments
> > I would avoid changing the structure of sections to ease mapping with 8776. For example, I would keep 1.1/1.2 as it was in 8776.

Italo: Ok, done

> > Acronyms: order them in an alphabetic order. Not sure the reasoning of the current ordering.

Italo: Ok, done

> > Include trees with -print-groupings or a pointer to where to retrieve the full trees.

Italo: added appendix with the generated YANG tree

> > The tree in 3.2 is not aligned with the YANG module: the bandwidth tree indicates that CIR/CBS are optional (with ?), while these are tagged as mandatory in the module.

Fixed

---

> > I have doubts about many of enums (e.g., path-type). Please check that these are justified vs using identities.

Most of the enumerations are from RFC 8776 and changing them would be an NBC change

The `path-type` should be an enumeration since it is follows the structure of the TE tunnel model. It can and shall be updated only when new types of paths are defined, through standard actions, in the TE tunnel model.

The 'te-generic-node-id' should also be an enumeration. It can and shall be updated only when new formatting for the `te-gen-node-id` data type are defined, through standard actions, in the TE types model .

---

> > MEF 6.2 EVC Ethernet Services Definitions Phase 3 - MEF : standard superseded. Please double check this reference.

I cannot find any reference to MEF 6.2

Originally posted by @italobusi in https://github.com/tsaad-dev/te/issues/279#issuecomment-2243651404_

### PR DESCRIPTION
**Pull request by italobusi**
Addressing additional WG LC comments:
- Addressing minor comments from Med not implemented in PR #280 : fix #290 
   - Added an appendix with YANG tree to RFC8776-bis: fix #290 
   - Reordered the acronyms and abbreviations in RFC8776-bis: fix #290 
- Updated bandwidth profile definitions: fix #285
-  Removed te-gen-node-id and moved union within the te-generic-node-id: fix #288 
- Fixed normative references: fix #287
- Clarify why path-computation-error-reason has not been defined in an IANA-maintained model: fix #281 
- Added formats for the MPLS-TE path bandwidth: fix #284

### CODE REVIEW

**Comment by italobusi** [comment]
@tsaad-dev : do we need the state container to contain only the signaled-bandwidth leaf?

What about the following change:

OLD

```
    +-- packet-bandwidth
       +-- specification-type?   te-bandwidth-requested-type
       ....................................................................................................................
       +-- class-type?           te-types:te-ds-class
       +-- state
          +-- signaled-bandwidth?   te-packet-types:bandwidth-kbps
```

NEW

```
    +-- packet-bandwidth
       +-- specification-type?   te-bandwidth-requested-type
       ....................................................................................................................
       +-- class-type?           te-types:te-ds-class
       +-- signaled-bandwidth?   te-packet-types:bandwidth-kbps
```

File: ietf-te-packet-types.yang [hunk 1]
```diff
@@ -767,17 +759,93 @@ module ietf-te-packet-types {
...
+      leaf class-type {
+        type te-types:te-ds-class;
+        description
+          "The Class-Type of traffic transported by the LSP.";
+        reference
+          "RFC 4124: Protocol Extensions for Support of
+                     Diffserv-aware MPLS Traffic Engineering,
+                     Section 4.3.1";
+      }
+      container state {
```
Comment (tsaad-dev) [inline ietf-te-packet-types.yang:817]: do we still use "state" container? Also, shouldn't this be config False?

File: ietf-te-packet-types.yang [hunk 2]
```diff
@@ -767,17 +759,93 @@ module ietf-te-packet-types {
...
   grouping te-packet-link-bandwidth {
     description
-      "Link Bandwidth for Packet. ";
+      "Bandwidth attributes for Packet links.";
     leaf packet-bandwidth {
       type uint64;
       units "bits/second";
       description
-        "Available bandwith value.";
+        "Bandwidth value for Packet links.";
```
Comment (tsaad-dev) [inline ietf-te-packet-types.yang:848]: s/Packet links/packet TE links/

**Review by tsaad-dev — APPROVED** [review]
please address couple of small nits

**Comment by italobusi** [comment]
**2024-09-27 TE Call**

Agreed to remove the state container and make the signaled-bandwidth config false (see also https://github.com/tsaad-dev/te/pull/291#discussion_r1777157979)


> @tsaad-dev : do we need the state container to contain only the signaled-bandwidth leaf?
> 
> What about the following change:
> 
> OLD
> 
> ```
>     +-- packet-bandwidth
>        +-- specification-type?   te-bandwidth-requested-type
>        ............................................................
>        +-- class-type?           te-types:te-ds-class
>        +-- state
>           +-- signaled-bandwidth?   te-packet-types:bandwidth-kbps
> ```
> 
> NEW
> 
> ```
>     +-- packet-bandwidth
>        +-- specification-type?   te-bandwidth-requested-type
>        ............................................................
>        +-- class-type?           te-types:te-ds-class
>        +-- signaled-bandwidth?   te-packet-types:bandwidth-kbps
> ```

File: ietf-te-packet-types.yang (hunk 1 above)
Comment (italobusi) [reply ietf-te-packet-types.yang:817]: Agreed.

See also: https://github.com/tsaad-dev/te/pull/291#issuecomment-2379389176

File: ietf-te-packet-types.yang [hunk 3]
```diff
@@ -767,17 +759,93 @@ module ietf-te-packet-types {
...
+  grouping te-packet-path-bandwidth {
+    description
+      "Bandwidth attributes for TE Packet paths.";
+    container packet-bandwidth {
+      description
+        "Bandwidth attributes for TE Packet paths.";
+      leaf specification-type {
+        type te-bandwidth-requested-type;
+        description
+          "The method used for setting the bandwidth, either
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-te-packet-types.yang]: Fixed in https://github.com/tsaad-dev/te/pull/291/commits/4b4632c6c61ab00d260a5171e979af5143999c54

File: ietf-te-packet-types.yang (hunk 1 above)
Comment (italobusi) [reply ietf-te-packet-types.yang]: Fixed in https://github.com/tsaad-dev/te/pull/291/commits/4b4632c6c61ab00d260a5171e979af5143999c54

File: ietf-te-packet-types.yang (hunk 2 above)
Comment (italobusi) [reply ietf-te-packet-types.yang]: Fixed in https://github.com/tsaad-dev/te/pull/291/commits/4b4632c6c61ab00d260a5171e979af5143999c54

**Pull request merged**